components/event-bus/internal/knative/subscription/controller/subscription/reconcile_test.go

```diff
@@ -366,8 +368,8 @@ func (k *MockKnativeLib) GetChannelByLabels(namespace string, labels map[string]
 	return k.GetChannel(channelName, namespace)
 }
 
-func (k *MockKnativeLib) CreateChannel(name string, namespace string, labels map[string]string, timeout time.Duration) (*messagingV1Alpha1.Channel, error) {
-	channel := makeKnChannel(namespace, name, labels)
+func (k *MockKnativeLib) CreateChannel(prefix, namespace string, labels map[string]string, timeout time.Duration) (*messagingV1Alpha1.Channel, error) {
```

Comment:
/begin note for the future (NOT this PR in particular)
All those mocks seem too far away from the actual implementation to be relied upon, and we would be better off using the Fake clientset against the real implementation instead.
/end

Reply:
i think, we use knative's unit testing f/w over here. for most of the reconciliation ops. however i agree to use fake client for knativelib operations as well. could you please create a ticket on this, @antoineco ?